homeassistant/components/screenlogic/__init__.py

```diff
@@ -124,8 +132,22 @@ async def _async_update_data(self):
         try:
             async with self.api_lock:
                 await self.hass.async_add_executor_job(self.gateway.update)
-        except ScreenLogicError as error:
-            raise UpdateFailed(error) from error
+        except ScreenLogicError:
+            _LOGGER.warning("ScreenLogicError - attempting reconnect")
+
+            try:
+                _LOGGER.warning("Re-connecting to the gateway")
+                connect_info = get_connect_info(self.hass, self.config_entry)
```

Comment:
Please only include the line(s) that can raise in the try... except block. Move the other lines outside the block.